File: Easy/Check if frequencies can be equal.cpp
//{ Driver Code Starts
#include <bits/stdc++.h>
using namespace std;


// } Driver Code Ends
//User function template for C++
class Solution{
public:	
	bool sameFreq(string s)
	{
	    // code here
	    int n = s.length();
	    int mappa[26] =  {0};
        for (int i=0; i<n; i++)
        {
            mappa[s[i] - 'a']++;
        }
        
        unordered_map<int,int> freq;
        for(int i=0; i<26; i++)
        {
            if(mappa[i] > 0)
            {
                freq[mappa[i]]++;
                if (freq.size() > 2) return false;
            }
        }
    
        if (freq.size() == 1) return true;
        else
        {
            auto it = freq.begin();
            auto it2 = it++;
            
            int k = it->second;
            int l = it2->second;
            
            if (l == 1 && ((it2->first-it->first == 1) || it2->first == 1)) return true;
            else if       (k==1 && ((it->first) - (it2->first) == 1 || it->first == 1)) return true;
            else          return false;
        }   
	}
};

//{ Driver Code Starts.
int main(){
    int t;
    cin>>t;
    while(t--)
    {
        string s;
        cin>>s;
        Solution ob;
        cout<<ob.sameFreq(s)<<endl;
    }
}



// } Driver Code Ends

File: Medium/Check if frequencies can be equal.py
#User function Template for python3
class Solution:
    def sameFreq(self, s):
        # code here
        d = {}
        
        for char in s:
            d[char] = d.get(char, 0) + 1

        d = list(d.values())
        
        dd = set(d)

        if len(dd) == 1:
            return True

        if len(dd) > 2:
            return False

        if d.count(max(d)) == 1 and max(d) - min(d) == 1:
            return True
        elif d.count(min(d)) == 1 and min(d) == 1:
            return True
            
        return False

        
        
        


#{ 
 # Driver Code Starts
#Initial Template for Python 3

if __name__ == "__main__":
	T=int(input())

	for _ in range(T):
		s = input()
		ob = Solution()
		answer = ob.sameFreq(s)
		if answer:
			print(1)
		else:
			print(0)

# } Driver Code Ends
